numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -1849,7 +1849,64 @@ BOOL_fromstr(char *str, void *ip, char **endptr,
 }
 
 /**begin repeat
- * #fname = CFLOAT, CDOUBLE, CLONGDOUBLE,
+ * #fname = CFLOAT, CDOUBLE#
+ * #type = npy_cfloat, npy_cdouble#
+ */
+static int
+@fname@_fromstr(char *str, void *ip, char **endptr,
+        PyArray_Descr *NPY_UNUSED(ignore))
+{
+    double result;
+
+    result = NumPyOS_ascii_strtod(str, endptr);
+
+    @type@ output;
+
+    if (endptr && ((*endptr[0] == '+') || (*endptr[0] == '-'))) {
+        // Imaginary component specified
+
+        output.real = result;
+
+        // Reading imaginary component
+        char **prev = endptr;
+        str = *endptr;
+        result = NumPyOS_ascii_strtod(str, endptr);
+
+        if (endptr && *endptr[0] == 'j') {
+            output.imag = result;
+        }
+        else {
+            endptr = prev;
```

Comment:
This is to indicate an error by not reading everything? I think it could even return a negative value in this case, in any case, I think I would like a comment explaining this path.

Reply:
Yeap. This is to trigger the error. I'm adding a comment on that. 

I'm fine with changing the imaginary part to -1 but can you explain why returning a negative is preferred since a negative number is equally likely as any other number? Thanks.

Reply:
I meant the integer return of the function, would have to check mysefl, but IIRC certain return values may also be accepted to signal error.